syntax = "proto3";
import "google/protobuf/empty.proto";

// to compile
// python3 -m grpc_tools.protoc -I./ --python_out=. --grpc_python_out=. --pyi_out=. ./meteo_utils.proto

package meteo;

message AirConditionParams {
  float temperature = 1;
  float co2 = 2;
  float humidity = 3;
}

message AirAnalysisResponse {
    float temperature = 1;
    float humidity = 2;
}
message PollutionAnalysisResponse {
    float co2 = 1;
}

message AirWellness {
  float wellness = 1;
}

service MeteoDataService {
  rpc GetAirConditionParams() returns (AirConditionParams) {}
  rpc CalculateAirWellness(AirConditionParams) returns (AirWellness) {}
  rpc CalculateAirWellness(AirConditionParams) returns (AirWellness) {}
  rpc CalculateAirWellness(AirConditionParams) returns (AirWellness) {}
  rpc CalculateAirWellness(AirConditionParams) returns (AirWellness) {}

  rpc ProcessMeteoData(MeteoData) returns (AirWellness) {}
  rpc ProcessPollutionData(PollutionData) returns (Co2Wellness) {}

  rpc AnalyzeAir() returns (AirAnalysisResponse) {}
  rpc AnalyzePollution() returns (PollutionAnalysisResponse) {}



analyze_air
}
